packages/peregrine/lib/talons/CartPage/GiftOptions/useGiftOptions.js

```diff
@@ -16,47 +15,41 @@ const useGiftOptions = props => {
     const [includeGiftReceipt, setIncludeGiftReceipt] = useState(false);
     const [includePrintedCard, setIncludePrintedCard] = useState(false);
     const [giftMessage, setGiftMessage] = useState('');
+    const apolloClient = useApolloClient();
 
     const [{ cartId }] = useCartContext();
 
-    const [fetchGiftOptions, { data }] = useLazyQuery(getGiftOptionsQuery, {
-        fetchPolicy: 'no-cache'
+    const { data } = useQuery(getGiftOptionsQuery, {
+        skip: !cartId,
+        variables: { cartId }
     });
 
-    /**
-     * Fetch gift options for a given cart id.
-     */
-    useEffect(() => {
-        if (cartId) {
-            fetchGiftOptions({
+    const updateGiftOptions = useCallback(
+        optionsToUpdate => {
+            apolloClient.cache.writeQuery({
```

Comment:
Resolvers are deprecated - no need to define a mutation when we can just write to the data referenced by the query.